crates/models/src/resources.rs

```diff
@@ -57,87 +68,153 @@ impl Import {
     fn example_extended() -> Self {
         Self::Extended {
             url: RelativeUrl::new("https://example/schema.json"),
-            content_type: ContentType::JsonSchema,
+            content_type: ContentType::JsonSchema(ContentFormat::Json),
+        }
+    }
+}
+
+/// ContentFormat describes the format for a resource.
```

Comment:
I don't believe this is needed and I'd like to see Json / Yaml distinctions of ContentFormat and json vs yaml-specific ContentType backed out of this PR.

The ContentType of a serialized JSON DOM can continue to simply be ContentType::Catalog or ContentType::JsonSchema. We'll parse the JSON as YAML, but that's totally fine and we already to that all the time.